app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsViewModel.kt

```diff
@@ -119,4 +139,37 @@ class AgendaDetailsViewModel @Inject constructor() : ViewModel() {
             )
         }
     }
+
+    fun onAddPhotosClick(photoUris: List<Uri>) {
+        _photosUri.value = photoUris
+    }
+
+    fun compressAndAddImage(context: Context, uris: List<Uri>) {
+        viewModelScope.launch {
+            var skipped = 0
+            val newCompressedList = uris.mapNotNull { uri ->
+                val drawable = imageCompressionUseCase.uriToDrawable(context, uri)
+                val originalBitmap = (drawable as BitmapDrawable).bitmap
+                Log.d(TAG, "Original Bitmap Size in bytes: ${originalBitmap.byteCount}")
+
+                val compressedByteArray = withContext(Dispatchers.IO) {
+                    imageCompressionUseCase.compressImage(drawable)
+                }
```

Comment:
As in my previous comment, I'd recommend to move this withContext block into the `compressImage` function itself. You added it here, which is good and won't cause any bugs in this context, but it violates the concept of main-safety (a suspend function should be safe to call from the main thread without blocking it). Just because you added the withContext block here doesn't mean that you'll always remember doing that when calling this function from somewhere else. If the withContext block is in the compressImage function however, it doesn't matter from which dispatcher you call it, since it makes sure by itself that it always runs on the right thread. More about that here:
https://www.youtube.com/watch?v=4_fDefmI3yI